### PR TITLE
ICU-22677 update LICENSE and README.md

### PR DESCRIPTION
**Pull request by srl295**
- https://github.com/unicode-org/.github/issues/15
- update year in LICENSE and README.md
- add SPDX link in LICENSE
- use SPDX code in pom.xml instead of raw license file

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22677
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [X] Issue accepted (done by Technical Committee after discussion)

### CODE REVIEW

**Comment by srl295** [comment]
similar activity but different scope from #2843

File: LICENSE
Comment (markusicu) [inline LICENSE:1]: Please update line 5 to say 2016-2024

Comment (srl295) [reply LICENSE:1]: thanks, missed that also

**Review by markusicu — APPROVED** [review]

**Comment by markusicu** [comment]
@srl295 please look at the CI failures

**Review by markusicu — APPROVED** [review]

**Pull request merged**